src/OrbitGl/ThreadTrack.cpp

```diff
@@ -501,8 +506,19 @@ void ThreadTrack::UpdatePrimitives(Batcher* batcher, uint64_t min_tick, uint64_t
         batcher->AddVerticalLine(pos, box_height, draw_data.z, color, std::move(user_data));
       }
 
-      // Use the time at boundary of the next pixel as a threshold to avoid overdraw.
-      next_pixel_start_time_ns = GetNextPixelBoundaryTimeNs(pos[0] + size[0], draw_data);
+      // Optimization when going through the scopes, as it doesn't have any sense to draw two timers
+      // in the same pixel.
+      // TODO (b/199736246): NextPixelBoundaryTimeNs should be greater than the pixel provided.
```

Comment:
Well, there are 2 different comments there.
 The first one is only a modification: Before we were using it only to avoid overdraw and now we are using it to also perform faster. 

The second one is a bug. NextPixelBoundaryTimeNs sometimes gave you a smaller timestamp (explained in the bug). I will try to write a better message.

What do you think?

Reply:
"Overdraw" is the correct terminology. https://developer.android.com/topic/performance/rendering/overdraw.

 Maybe put the TODO in the GetNextPixelBoundaryTimeNs function.

Reply:
For info "doesn't have any sense" should be "doesn't *make* any sense", but I'd avoid such phrasing in comments, it seems a bit negative to me.

Reply:
Thanks for your comments, will address them tomorrow :)

Reply:
The TODO has been addressed and this comment is not present anymore.